resources/knative-eventing-kafka/pre-upgrade.sh

```diff
@@ -0,0 +1,5 @@
+#!/usr/bin/env bash
+
+set -eux -o pipefail
+
+kubectl delete deployment -n knative-eventing -lapp.kubernetes.io/instance=knative-eventing-kafka --ignore-not-found
```

Comment:
Should we template the script as well? 🤔 

```suggestion
kubectl delete deployment -n {{ .Release.Namespace }} -lapp.kubernetes.io/instance={{ .Release.Name }} --ignore-not-found
```

Reply:
Scripts cannot be templated. The placeholders are rendered as in. For example:
```
.....
          command:
            - "/bin/bash"
          args:
            - "-c"
            - |
              #!/usr/bin/env bash

              set -eux -o pipefail

              kubectl delete deployment -n {{ .Release.Namespace }} -lapp.kubernetes.io/instance={{ .Release.Name }} --ignore-not-found
          terminationMessagePolicy: "FallbackToLogsOnError"
          resources:
            requests:
              cpu: 200m
              memory: 128Mi
            limits:
              cpu: 200m
              memory: 128Mi
```

Reply:
if we would inline this script into the yaml, we could template it... so the question would be. does it make sense?

Reply:
We could but that would make this inconsistent with the other helm hook jobs. E.g. uninstall

Reply:
Sad, I hoped it would work without inlining. We idea of having a script which is included in the job was that you can run it locally. This would not be the case anymore with templating. Don't have a strong opinion here, continue as you want from my side :)